AGENTS.md

```diff
@@ -374,71 +476,318 @@ isort --check-only src/ tests/
 black --check src/ tests/
 ```
 
+### Unused Imports Detection
+
+**CRITICAL**: Always verify there are no unused imports before committing code.
+
+```bash
+# Use autoflake to detect unused imports
+autoflake --check --remove-all-unused-imports src/ tests/
+
+# Or use your IDE's built-in detection
+# PyCharm: Code -> Optimize Imports
+# VSCode: Organize Imports command
+```
+
+#### Common Unused Import Scenarios
+
+```python
+# ❌ FORBIDDEN: Unused imports left in code
+from typing import Dict, List, Optional  # Optional not used
+import os  # Not used (and should use adaptor anyway!)
+from dd_license_attribution.utils import helper  # Not used
+
+def process(data: list[str]) -> dict[str, str]:
+    return {}
+
+# ✅ REQUIRED: Only import what you use
+from typing import Any
```

Comment:
will it be obvious to the AI that this line and line 506 are patterns to NOT use without you labeling it that way?